src/Raven.Server/Documents/Patch/PatchDocumentCommand.cs

```diff
@@ -461,24 +467,43 @@ public class PatchDocumentCommand : PatchDocumentCommandBase
             _ignoreMaxStepsForScript = ignoreMaxStepsForScript;
             if (string.IsNullOrEmpty(id) || id.EndsWith(identityPartsSeparator) || id.EndsWith('|'))
                 throw new ArgumentException($"The ID argument has invalid value: '{id}'", nameof(id));
+
+            // If the caller is a DocumentsOperationContext, then we can apply the optimization.
```

Comment:
What _else_ can this be?

Reply:
As far back as I could go. Apparently it is only a `DocumentOperationContext`. Let me check if I can restrict the callers along the way to only support that.

Reply:
Ok in cases of sharded databases we can end up getting a `TransactionOperationContext` at creation which does not hold a `DocumentDatabase` instance. So, we need to defer it.